javaslang/src/main/java/javaslang/collection/LinkedHashMultimap.java

```diff
@@ -0,0 +1,162 @@
+ * @since 2.0.0
+ */
+public class LinkedHashMultimap<K, V> extends AbstractMultimap<K, V, LinkedHashMultimap<K, V>> implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    public static <V> Builder<V> withSeq() {
+        return new Builder<>(ContainerType.SEQ, List::empty);
+    }
+
+    public static <V> Builder<V> withSet() {
+        return new Builder<>(ContainerType.SET, HashSet::empty);
+    }
+
+    public static <V extends Comparable<? super V>> Builder<V> withSortedSet() {
+        return new Builder<>(ContainerType.SORTED_SET, TreeSet::empty);
+    }
+
+    public static <V> Builder<V> withSortedSet(Comparator<? super V> comparator) {
+        return new Builder<>(ContainerType.SORTED_SET, () -> TreeSet.empty(comparator));
+    }
+
+    public static class Builder<V> {
+
+        private final ContainerType containerType;
+        private final SerializableSupplier<Traversable<?>> emptyContainer;
+
+        private Builder(ContainerType containerType, SerializableSupplier<Traversable<?>> emptyContainer) {
+            this.containerType = containerType;
+            this.emptyContainer = emptyContainer;
+        }
+
+        public <K, V2 extends V> LinkedHashMultimap<K, V2> empty() {
+            return new LinkedHashMultimap<>(LinkedHashMap.empty(), containerType, emptyContainer);
+        }
+
+        public <K, V2 extends V> LinkedHashMultimap<K, V2> ofEntries(Iterable<? extends Tuple2<? extends K, ? extends V2>> entries) {
+            Objects.requireNonNull(entries, "entries is null");
+            Multimap<K, V2> result = empty();
+            for (Tuple2<? extends K, ? extends V2> entry : entries) {
+                result = result.put(entry._1, entry._2);
+            }
+            return (LinkedHashMultimap<K, V2>) result;
+        }
+
+        @SafeVarargs
+        public final <K, V2 extends V> LinkedHashMultimap<K, V2> ofEntries(Tuple2<? extends K, ? extends V2>... entries) {
+            Objects.requireNonNull(entries, "entries is null");
+            Multimap<K, V2> result = empty();
+            for (Tuple2<? extends K, ? extends V2> entry : entries) {
+                result = result.put(entry._1, entry._2);
+            }
+            return (LinkedHashMultimap<K, V2>) result;
+        }
+
+        @SafeVarargs
+        public final <K, V2 extends V> LinkedHashMultimap<K, V2> ofEntries(java.util.Map.Entry<? extends K, ? extends V2>... entries) {
+            Objects.requireNonNull(entries, "entries is null");
+            Multimap<K, V2> result = empty();
+            for (java.util.Map.Entry<? extends K, ? extends V2> entry : entries) {
+                result = result.put(entry.getKey(), entry.getValue());
+            }
+            return (LinkedHashMultimap<K, V2>) result;
+        }
+
+        @SuppressWarnings("unchecked")
+        public <K, V2 extends V> LinkedHashMultimap<K, V2> tabulate(int n, Function<? super Integer, ? extends Tuple2<? extends K, ? extends V2>> f) {
```

Comment:
Dito... I will not mark the other ? extends Tuple occurrences. I'm sure I'll miss one. Would be best to make search-replace.